docs_src/microservices/security/Ch-SSH-Tunneling-HowToSecureDeviceServices.md

```diff
@@ -0,0 +1,297 @@
+
+[docker-compose file for remote device services with SSH server/daemon](https://github.com/edgexfoundry-holding/external-device-security-examples/ssh-tunneling/edgex-device-sshd-remote.yml)
+
+Note that the following ssh server service is added in the docker-compose file:
+
+```yaml
+################################################################
+# SSH Daemon
+################################################################
+  sshd-remote:
+    image: eg_sshd
+    ports:
+      - "2223:22"
+    container_name: edgex-sshd-remote
+    hostname: edgex-sshd-remote
+    networks:
+      edgex-network:
+        aliases:
+        - edgex-core-consul
+        - edgex-core-data
+        - edgex-core-metadata
+```
+
+- In the primary machine, include `device-ssh-proxy:test` ssh proxy docker image together with EdgeX core services in the docker-compose file like this:
+
+```yaml
+##########################################################
+# ssh tunneling proxy service for device-virtual
+##########################################################
+  device-ssh-proxy:
+    image: device-ssh-proxy:test
+    container_name: edgex-device-ssh-proxy
+    hostname: edgex-device-ssh-proxy
+    volumes:
+      - $HOME/.ssh:/root/ssh:ro
+    ports:
+      - "49990:49990"
+    networks:
+      edgex-network:
+        aliases:
+            - edgex-device-virtual
+    environment:
+      TUNNEL_HOST: 192.168.1.190
+      TUNNEL_SSH_PORT: 2223
+      SERVICE_HOST: edgex-device-virtual
+      SERVICE_PORT: 49990
+```
+
+The full docker-compose file is included here:
+
+[docker-compose file for the primary core services and ssh tunneling proxy service without any device services](https://github.com/edgexfoundry-holding/external-device-security-examples/ssh-tunneling/edgex-core-ssh-proxy.yml)
+
+Note that:
+
+1. The values of environment variables depend on your environment settings of the primary and the remote machine. In this particular case, we are ssh tunneling to the remote device-virtual service.
+
+2. The docker-compose file in the primary machine does not include any device services at all.  This is to ensure that we are actually using the device services in the remote machine.
+
+#### Test with the device-virtual APIs
+
+- mainly run curl or postman directly from the primary machine to the device-virtual APIs to verify the remote device virtual service can be accessible from the primary host machine via two-way SSH tunneling. This can be checked from the console of the primary machine:
+
+the ping response of calling edgex-device-virtual's ping action:
+
+```sh
+jim@jim-NUC7i5DNHE:~/go/src/github.com/edgexfoundry/developer-scripts/releases/geneva/compose-files$ curl http://localhost:49990/api/v1/ping
+
+1.2.0-dev.13j
+
+```
+
+or see the configuration of it:
+
+```sh
+jim@jim-NUC7i5DNHE:~/go/src/github.com/edgexfoundry/developer-scripts/releases/geneva/compose-files$ curl http://localhost:49990/api/v1/config
+
```

Comment:
create a separate code block for this formatted for JSON

Reply:
added